src/Components/ServiceScene/Breakdowns/AddToFiltersButton.tsx

```diff
@@ -19,17 +20,24 @@ export interface AddToFiltersButtonState extends SceneObjectState {
 
 export class AddToFiltersButton extends SceneObjectBase<AddToFiltersButtonState> {
   public onClick = () => {
-    const variable = sceneGraph.lookupVariable(this.state.variableName, this);
-    if (!(variable instanceof AdHocFiltersVariable)) {
-      return;
-    }
-
     const labels = this.state.frame.fields[1]?.labels ?? {};
     if (Object.keys(labels).length !== 1) {
       return;
     }
     const labelName = Object.keys(labels)[0];
 
+    let variableName = this.state.variableName;
+    // If the variable is a level variable, we need to use the VAR_FIELDS variable
+    // as that one is detected field
+    if (labelName === LEVEL_VARIABLE_VALUE) {
+      variableName = VAR_FIELDS;
+    }
+
+    const variable = sceneGraph.lookupVariable(variableName, this);
+    if (!(variable instanceof AdHocFiltersVariable)) {
+      return;
```

Comment:
Is this runtime check just to keep the types from complaining? Do we want to output a console.warn so we can more easily track down a situation (even in dev in the future) where we fail to grab the variable?

Reply:
fixed in https://github.com/grafana/explore-logs/pull/432/commits/a494ec09054ce46d58c53809e466ae8d864666bc